rust/agama-server/src/storage/web.rs

```diff
@@ -109,6 +110,55 @@ pub async fn storage_service(dbus: zbus::Connection) -> Result<Router, ServiceEr
     Ok(router)
 }
 
+/// Returns the storage configuration.
+///
+/// * `state` : service state.
+#[utoipa::path(
+    get,
+    path = "/config",
+    context_path = "/api/storage",
+    operation_id = "get_storage_config",
+    responses(
+        (status = 200, description = "storage configuration", body = StorageSettings),
+        (status = 400, description = "The D-Bus service could not perform the action")
+    )
+)]
+async fn get_config(State(state): State<StorageState<'_>>) -> Result<Json<StorageSettings>, Error> {
+    // StorageSettings is just a wrapper over serde_json::value::RawValue
+    let settings = state
+        .client
+        .get_config()
+        .await
+        .map_err(|e| Error::Service(e))?;
+    Ok(Json(settings))
+}
+
+/// Sets the storage configuration.
+///
+/// * `state`: service state.
+/// * `config`: storage configuration.
+#[utoipa::path(
+    put,
+    path = "/config",
+    context_path = "/api/storage",
+    operation_id = "set_storage_config",
+    responses(
+        (status = 200, description = "Set the storage configuration"),
+        (status = 400, description = "The D-Bus service could not perform the action")
+    )
+)]
+async fn set_config(
+    State(state): State<StorageState<'_>>,
+    Json(settings): Json<StorageSettings>,
+) -> Result<Json<()>, Error> {
+    let _status: u32 = state
+        .client
+        .set_config(settings)
+        .await
+        .map_err(|e| Error::Service(e))?;
```

Comment:
You can write this as:

```suggestion
        .map_err(Error::Service)?;
```